src/Model/Behavior/SearchBehavior.php

```diff
@@ -214,10 +214,13 @@ protected function _processFilters($filters, $params, $query)
             $filter->args($params);
             $filter->query($query);
 
-            if (!$filter->skip()) {
+            if ($filter->skip()) {
+                continue;
+            }
+            $result = $filter->process();
+            if ($result !== false) {
```

Comment:
Ah right it checks for not false. This is why i dislike negations in general, easy to misread :) Valid use here.